### PR TITLE
Update to latest nightly, cargo dep update and remove workflow hacks

### PR DESCRIPTION
**Pull request by jmt-lab**
**Description of changes:**

* updates nightly to 2024-07-11 so new used features in serde exist
* updates cargo dependencies to ensure time is updated to a suported version for the nightly
* removes previous workarounds in workflows

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by sam-berning — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
           submodules: recursive
       - name: Install cargo-dist
         run: |
-          cargo install --locked \
+          cargo install \
             --git "https://github.com/webern/cargo-dist" \
```
Comment (cbgbt) [inline .github/workflows/release.yml]: Locking dependencies in these cases seems like a good idea to me. Should we really be removing this?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -22,6 +22,6 @@ jobs:
       labels: bottlerocket_ubuntu-latest_16-core
     steps:
       - uses: actions/checkout@v3
-      - run: cargo install --locked cargo-deny
-      - run: cargo install --locked cargo-make@0.37.12
```
Comment (cbgbt) [inline .github/workflows/rust.yml:26]: I think we originally locked to `0.37.12` for `cargo-make` due to a tendency to introduce breaking changes upstream. Might be worth continuing to require a specific version.

**Review by cbgbt — APPROVED** [review]
This mostly LGTM, though I think we will be happier if we maintain a versioned dependency on `cargo-make`, even if the version changes.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (jmt-lab) [reply .github/workflows/rust.yml:26]: Actually in this case I added this version lock before realizing that we needed to bump nightly. The newer patch of cargo-make would pull in the serde version that was incompatible with the old nightly

Comment (cbgbt) [reply .github/workflows/rust.yml:26]: Ah, I just misremembered. Forget my feedback then! 🚀

**Pull request merged**